drivers/ioexpander/tca64xx.c

```diff
@@ -439,6 +482,55 @@ static int tca64_direction(FAR struct ioexpander_dev_s *dev, uint8_t pin,
               regaddr, ret);
     }
 
+  if ((direction == IOEXPANDER_DIRECTION_IN_PULLDOWN) ||
+     (direction == IOEXPANDER_DIRECTION_IN_PULLUP))
+    {
+      regaddr = tca64_pdenable_reg(priv, pin);
```

Comment:
There is no error handling now if PU/PD setting is attempted for other parts than the one which has the register. You should propably add "define INVALID_REG  0xff" or such, and make tca64_pdenable_reg return that if part->tp_pu_select is 0.

Then you can check the regaddr here and bail out if the part doesn't support PU/PD....

Reply:
It should be fixed now.